### ISSUE TITLE
Since 1.2.0 json responses from http api messages are escaped

### ISSUE THREAD

**Issue by w4tsn** [eclipse/ditto#805]
I've updated ditto 1.1.x to 1.2.1 and since then I'm observing json responses of ditto messages being escaped where they were not before the update.

I've a micro service which is reachable through dittos messaging API via HTTP calls using the WebSocket API. The service answers the calls with a JSON object. The object is stringyfied before it's returned as a response. A `'content-type': 'application/json'` is provided with the response.

```
POST /api/2/things/test:3AF63B05432A65C32A/features/Lamp/inbox/messages/unit
``` 

Before the update the response would just be a stringyfied JSON string, since 1.2 it's escaped.

```
# Before 1.2
{"feature":"Lamp","property":"active","measurementUnit":"state"}

# Since 1.2
"{\"feature\":\"Lamp\",\"property\":\"active\",\"measurementUnit\":\"state\"}"
```

I've not tested it with other content types than application/json.

EDIT: I've now downgraded from 1.2.1 to 1.1.5 and can verify that the behavior returned back to normal / that this change is introduced somewhere in 1.2 or 1.2.1.

**Comment by thjaeckle** [comment]
We can confirm that this is a bug.
The fix is part of PR #809 
We expect to do a Ditto 1.3.0 release in the upcoming week(s) in order to get a fix out soon.

**Comment by w4tsn** [comment]
Just in case this takes longer than this week I'd actually expect a patch release 1.2.2 since this is a breaking change which is blocking us from upgrading to 1.2.x entirely.

Since this problem did not get caught earlier do you know if it's rather unusual how we use ditto protocol messages in micro services connected through websockets?

**Comment by thjaeckle** [comment]
> I'd actually expect a patch release 1.2.2 since this is a breaking change which is blocking us from upgrading to 1.2.x entirely.

There won't be a backport of the bugfix to 1.2 as this would cause too much overhead and merge effort.

> Since this problem did not get caught earlier do you know if it's rather unusual how we use ditto protocol messages in micro services connected through websockets?

It's a quite advanced "pattern" which is not used too widely yet.
Please also keep in mind that a commercial offering might bring a bugfix sooner into production than the fix being available in a public Ditto release.

**Comment by thjaeckle** [comment]
Fixed by #809

**Comment by w4tsn** [comment]
@thjaeckle after the upgrade to 1.3 today, I'm still facing this problem. The response is still escaped. I'm not yet sure why and have to take a deeper look to give you even more information.

Our websocket connected service is sending it's response with a `"content-type": "application/json"` response.

Is it possible that something changed in the way that messages and responses are processed, that has to be configured now and differs from the implicit default behavior of ditto < 1.2? In 1.2 and 1.3 a lot changed in the area of messages, so do I have to use special headers or anything now?

Also what else could help you debug this?

**Comment by thjaeckle** [comment]
I'll try to reproduce on Monday when I'm back in the office.

**Comment by thjaeckle** [comment]
Ok, just tried the "happy path" on the Ditto Sandbox:

### Subscribed to WS for messages

* open https://www.websocket.org/echo.html
* connect to `wss://ditto:ditto@ditto.eclipse.org/ws/2`
* send message `START-SEND-MESSAGES`

### HTTP request
```
POST https://ditto:ditto@ditto.eclipse.org/api/2/things/org.eclipse.ditto:test-message-response/inbox/messages/hello?timeout=1m
```
Headers:
* Content-Type: application/json

Body:
```json
{
    "json": "value"
}
```

### Receive message in WS session

```json
{
	"topic": "org.eclipse.ditto/test-message-response/things/live/messages/hello",
	"headers": {
		"x-ditto-pre-authenticated": "nginx:ditto",
		"correlation-id": "2a84dba5-df72-48d9-86e6-2d16910ed38e",
		"x-forwarded-for": "194.39.218.10",
		"version": 2,
		"timeout": "1m",
		"accept": "*/*",
		"authorization": "Basic ZGl0dG86ZGl0dG8=",
		"x-real-ip": "194.39.218.10",
		"x-forwarded-user": "ditto",
		"response-required": true,
		"ditto-originator": "nginx:ditto",
		"host": "ditto.eclipse.org",
		"content-type": "application/json",
		"accept-encoding": "gzip, deflate, br",
		"timestamp": "2020-10-05T12:09:03.960541112+02:00",
		"user-agent": "PostmanRuntime/7.26.5"
	},
	"path": "/inbox/messages/hello",
	"value": {
		"json": "value"
	}
}
```

### Answer via WS message

```json
{
	"topic": "org.eclipse.ditto/test-message-response/things/live/messages/hello",
	"headers": {
		"correlation-id": "2a84dba5-df72-48d9-86e6-2d16910ed38e",
		"version": 2,
		"content-type": "application/json"
	},
	"path": "/inbox/messages/hello",
	"value": {
		"my response": "is this"
	},
	"status": 200
}
```

### Received HTTP response

Headers:
* Content-Type: application/json
* correlation-id: 2a84dba5-df72-48d9-86e6-2d16910ed38e
* ...

Body:
```json
{
    "my response": "is this"
}
```

## Unhappy path

The only way I can reproduce is by sending an escaped JSON as "value":

```json
{
	"topic": "org.eclipse.ditto/test-message-response/things/live/messages/hello",
	"headers": {
		"correlation-id": "239b383f-fc09-48e1-86e2-70477fc71e88",
		"version": 2,
		"content-type": "application/json"
	},
	"path": "/inbox/messages/hello",
	"value": "\"my response\": \"is this\"",
	"status": 200
}
```

This leads - as expected - to the HTTP response:
```json
"\"my response\": \"is this\""
```

@w4tsn could you re-check what you are sending as message response? Maybe there were changes in Ditto which now handle this "correctly" and cause the escaped JSON string - but more as a bug of an old Ditto version which is fixed since 1.2.0.

**Comment by w4tsn** [comment]
@thjaeckle as you said the problem was that we adopted a bug while working with ditto < 1.2.0. For some reason we changed responses from plain JSON to it's unescaped stringified representation a while ago. Since the content-type was correct and ditto did nothing special with the response it worked. Now it does escape string type responses which is fair, but breaking with our micro-services. After changing back to plain JSON responses everything is fine now.
